lib/Service/BotService.php

```diff
@@ -0,0 +1,285 @@
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+namespace OCA\Talk\Service;
+
+use OCA\Talk\Chat\MessageParser;
+use OCA\Talk\Events\ChatEvent;
+use OCA\Talk\Events\ChatParticipantEvent;
+use OCA\Talk\Model\Attendee;
+use OCA\Talk\Model\Bot;
+use OCA\Talk\Model\BotConversation;
+use OCA\Talk\Model\BotConversationMapper;
+use OCA\Talk\Model\BotServerMapper;
+use OCA\Talk\Room;
+use OCA\Talk\TalkSession;
+use OCP\AppFramework\Http;
+use OCP\AppFramework\Utility\ITimeFactory;
+use OCP\Http\Client\IClientService;
+use OCP\Http\Client\IResponse;
+use OCP\IConfig;
+use OCP\ISession;
+use OCP\IURLGenerator;
+use OCP\IUser;
+use OCP\IUserSession;
+use OCP\L10N\IFactory;
+use OCP\Security\ISecureRandom;
+use Psr\Log\LoggerInterface;
+
+class BotService {
+	public function __construct(
+		protected BotServerMapper       $botServerMapper,
+		protected BotConversationMapper $botConversationMapper,
+		protected IClientService        $clientService,
+		protected IConfig               $serverConfig,
+		protected IUserSession          $userSession,
+		protected TalkSession           $talkSession,
+		protected ISession              $session,
+		protected ISecureRandom         $secureRandom,
+		protected IURLGenerator         $urlGenerator,
+		protected IFactory              $l10nFactory,
+		protected ITimeFactory          $timeFactory,
+		protected LoggerInterface       $logger,
+	) {
+	}
+
+	public function afterChatMessageSent(ChatParticipantEvent $event, MessageParser $messageParser): void {
+		$bots = $this->getBotsForToken($event->getRoom()->getToken());
+		if (empty($bots)) {
+			return;
+		}
+
+		$message = $messageParser->createMessage(
+			$event->getRoom(),
+			$event->getParticipant(),
+			$event->getComment(),
+			$this->l10nFactory->get('spreed', 'en', 'en')
+		);
+		$messageParser->parseMessage($message);
+		$messageData = [
+			'message' => $message->getMessage(),
+			'parameters' => $message->getMessageParameters(),
+		];
+
+		$attendee = $event->getParticipant()->getAttendee();
+
+		$this->sendAsyncRequests($bots, [
+			'type' => 'Create',
+			'actor' => [
+				'type' => 'Person',
+				'id' => $attendee->getActorType() . '/' . $attendee->getActorId(),
+				'name' => $attendee->getDisplayName(),
+			],
+			'object' => [
+				'type' => 'Note',
+				'id' => $event->getComment()->getId(),
+				'name' => 'message',
+				'content' => json_encode($messageData),
```

Comment:
I think the important one is `JSON_THROW_ON_ERROR` just in case you have invalid data. Others only saves few bytes: 
```suggestion
				'content' => json_encode($messageData, JSON_UNESCAPED_LINE_TERMINATORS|JSON_UNESCAPED_SLASHES|JSON_UNESCAPED_UNICODE|JSON_THROW_ON_ERROR),
```
It is also possible to check the result of `json_encode`
(same for others `json_encode` calls)